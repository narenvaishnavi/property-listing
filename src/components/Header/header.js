import { Navbar } from 'react-bootstrap';
import './header.scss';

export default function Header({ title }) {
  return (
    <Navbar fixed="top" expand="lg" variant="light" bg="light">
      <div className="header" id="myHeader">
        <h2>{title}</h2>
      </div>
    </Navbar>
  );
}
