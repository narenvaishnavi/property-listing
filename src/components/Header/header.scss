@import '../../styles/global.scss';

.header {
  height: 103px;
  background: $headerBackground;
  border-radius: 1px;
  position: fixed;
  top: 0;
  width: 100%;
}

h2 {
  height: 40px;
  width: 294.93;
  left: -480px;
  top: -489.5px;
  font-family: $fontFamily;
  font-size: 28px;
  font-style: normal;
  font-weight: 600;
  line-height: 40px;
  letter-spacing: 0px;
  text-align: left;
  color: $headerTitleColor;
  padding: 1.5rem;
}
