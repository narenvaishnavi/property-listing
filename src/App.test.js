import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Card from './components/card/card';
import Header from './components/Header/header';

Enzyme.configure({ adapter: new Adapter() });

describe('Property List', () => {
  const mockData = [
    {
      id: 1,
      bedroom: '2BR',
      baths: 3.5,
      area: 2700,
      listPrice: 950000,
      address: '456 Avenue Crt',
      listDate: '01/02/2021',
      photo:
        'https://s3-us-west-2.amazonaws.com/cdn.simplyrets.com/properties/trial/home-inside-5.jpg',
      favorite: false,
    },
  ];
  const handleClick = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Card updateEstate={mockData} parentCallback={handleClick} />,
    );
  });

  it('Data Passed to CardList is not Null', () => {
    expect(wrapper).not.toBeNull();
  });

  it('Data Passed to CardList is not an Object', () => {
    expect(wrapper).not.toBe({});
  });

  it('Data Passed to CardList is an array of object', () => {
    expect(mockData).toEqual(
      expect.arrayContaining([expect.objectContaining({ id: 1 })]),
    );
  });

  it('Simulate HandleClick should always return object', () => {
    wrapper.find(Card).first().simulate('click');
    expect(wrapper).toEqual({});
  });

  it('Card Component render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Header Component render correctly', () => {
    wrapper = mount(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
