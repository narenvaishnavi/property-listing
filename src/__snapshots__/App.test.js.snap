// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property List Card Component render correctly 1`] = `ReactWrapper {}`;

exports[`Property List Header Component render correctly 1`] = `ReactWrapper {}`;
